src/pip/_internal/req/constructors.py

```diff
@@ -434,6 +434,21 @@ def install_req_from_req_string(
             "{} depends on {} ".format(comes_from.name, req)
         )
 
+    if req.url:
+        # Create an InstallRequirement for a wheel-like PEP 508 URL with the
+        # same behavior as 'pip install req.url'
+        parts = parse_req_from_line(req.url, None)
+        link = Link(req.url)
+        if link.is_wheel:
```

Comment:
I dropped support for directories and non-wheel URLs in 2e082f916ada16ba94b851f6cb9118f3986d1ba0 because that was triggering an assertion error in the resolver. Hopefully this PR can be a stopgap until https://github.com/pypa/pip/issues/7678 can be addressed in the new resolver?